src/test/java/org/candlepin/resource/ConsumerResourceCreationTest.java

```diff
@@ -259,7 +294,7 @@ public void init() throws Exception {
         when(ownerCurator.getByKey(owner.getKey())).thenReturn(owner);
         when(complianceRules.getStatus(
             any(Consumer.class), any(Date.class), any(Boolean.class), any(Boolean.class)))
-                .thenReturn(new ComplianceStatus(new Date()));
+            .thenReturn(new ComplianceStatus(new Date()));
```

Comment:
If a linter told you to remove that indentation, it was wrong. If anything, this should be re-written as a `doReturn(...).when(...)` type of mock, as it might help with readability in both testing and as far as the linter goes.

Reply:
Checkstyle does not support this. If line starts at column 8, it requires all of its continuation lines to start at column 12. Until we move to spotless we are stuck with this format.

Regarding doReturn. While I agree that doReturn is great for readability, I would advise against preferring it over when-then. Mainly it's not type-safe. when-then is less convenient, but does compile-time type checking which saves a ton of work when refactoring.

Reply:
I'd much rather have a `@SuppressWarnings("indentation")` than awful looking syntax, but as far as the doReturn/when thing goes, neither method is "safe" enough to justify only using one or the other. Both have trade-offs [1] in terms of how they function or how they read, and being able to use both is valuable for cases like this, where the actual setup we use isn't significant, and we have something else driving a largely unnecessary code change.

As a slight aside, while a compile-time failure *is* nice, if such an unchecked error hits a PR, it suggests that the test wasn't actually run before the PR was created.

[1] `when` behaves strangely with certain mocks/spys as it's making an actual call to the underlying mocked method during this setup call. Compatibility with the answer syntax and sequenced returns is also different between the two.

Reply:
Good point. suppression will let us format code better until we can get formatter to do that for us.

Running the PR on Jenkins is not an issue I was trying to point out. With `doReturn`, when we refactor some code. Like type of some param. We won't know whether it broke something until we run the tests. With `when.thenReturn`, we will know immediately because the code won't compile. Yes, there are places where when is not a suitable choice, but that does not mean it shouldn't be used at all.